src/components/TodoItem.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useContext, useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../type/Todo';
+import { TodosContext } from './TodosContext';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { dispatch } = useContext(TodosContext);
+  const [editId, setEditId] = useState(0);
```

Comment:
```suggestion
  const [isEditing, setIsEditing] = useState(false);
```
looks like it can be the boolean state